File: Projects/Python/Beginner-Projects/calculator/calculator-2/calculator.py
# Welcome to Level 2! 
# Now that you know the basics, we're going to organize our code better.
# In professional coding, we don't just write one big list of instructions. 
# We group them into "Functions" (like recipes) that we can call whenever we need them.

# --- THE RECIPES (FUNCTIONS) ---

# This function takes two numbers (x and y) and returns their sum.
def add(x, y):
    return x + y

# Subtraction recipe
def subtract(x, y):
    return x - y

# Multiplication recipe
def multiply(x, y):
    return x * y

# Division recipe - notice we're still checking for that pesky zero!
def divide(x, y):
    if y == 0:
        return "[E] Cannot divide by zero!"
    return x / y

# --- THE MAIN BRAIN ---

def main():
    # This is where our program actually 'starts'. 
    # By putting it in a function called main(), we keep things tidy.
    
    print("=== THE MODULAR CALCULATOR ===")
    print("This version uses FUNCTIONS to do the hard work.\n")

    # Let's try to get numbers, but we'll use a 'try' block.
    # If a user types 'abc' instead of a number, the program won't crash!
    try:
        num1 = float(input("First number: "))
        num2 = float(input("Second number: "))
    except ValueError:
        print("Wait! That's not a number. Please try again with actual numbers.")
        return # This stops the function right here.

    print("Pick one: +, -, *, /")
    op = input("Operation: ")

    # Now we call our 'recipes' based on what the user picked.
    if op == "+":
        print(f"Result: {add(num1, num2)}")
    elif op == "-":
        print(f"Result: {subtract(num1, num2)}")
    elif op == "*":
        print(f"Result: {multiply(num1, num2)}")
    elif op == "/":
        print(f"Result: {divide(num1, num2)}")
    else:
        print("Invalid choice. Please run me again and use +, -, *, or /.")

# This bit at the bottom is very important in Python.
# it basically says: "Only run main() if this file is the one we clicked 'Run' on."
# It stops the program from running automatically if we try to 'import' this file elsewhere.
if __name__ == "__main__":
    main()

File: Projects/C/Beginner-Projects/calculator/calc-2/calculator.c
#include <stdio.h>
#include <stdlib.h>

// Function prototypes
double add(double a, double b);
double subtract(double a, double b);
double multiply(double a, double b);
double divide(double a, double b);
void displayMenu();

int main() {
  int choice;
  double num1, num2, result;

  do {
    displayMenu();
    printf("Enter your choice (1-5): ");
    scanf("%d", &choice);

    if (choice >= 1 && choice <= 4) {
      printf("Enter two numbers: ");
      scanf("%lf %lf", &num1, &num2);
    }

    switch (choice) {
    case 1:
      result = add(num1, num2);
      printf("Result: %.2lf\n", result);
      break;
    case 2:
      result = subtract(num1, num2);
      printf("Result: %.2lf\n", result);
      break;
    case 3:
      result = multiply(num1, num2);
      printf("Result: %.2lf\n", result);
      break;
    case 4:
      result = divide(num1, num2);
      printf("Result: %.2lf\n", result);
      break;
    case 5:
      printf("Exiting calculator. Goodbye!\n");
      break;
    default:
      printf("Invalid choice! Please try again.\n");
    }

    printf("\n");
  } while (choice != 5);

  return 0;
}

void displayMenu() {
  printf("=== MODULAR CALCULATOR ===\n");
  printf("1. Addition\n");
  printf("2. Subtraction\n");
  printf("3. Multiplication\n");
  printf("4. Division\n");
  printf("5. Exit\n");
}

double add(double a, double b) { return a + b; }

double subtract(double a, double b) { return a - b; }

double multiply(double a, double b) { return a * b; }

double divide(double a, double b) {
  if (b != 0) {
    return a / b;
  } else {
    printf("Error: Division by zero!\n");
    return 0;
  }
}
